test/test_prototype_transforms.py

```diff
@@ -152,3 +153,56 @@ def test_normalize(self, transform, input):
     )
     def test_random_resized_crop(self, transform, input):
         transform(input)
+
+
+class TestRandomHorizontalFlip:
+    def input_tensor(self, dtype=torch.float32):
+        return torch.tensor([[[0, 1], [0, 1]], [[1, 0], [1, 0]]], dtype=dtype)
+
+    def expected_tensor(self, dtype=torch.float32):
+        return torch.tensor([[[1, 0], [1, 0]], [[0, 1], [0, 1]]], dtype=dtype)
+
+    @pytest.mark.parametrize("p", [0.0, 1.0], ids=["p=0", "p=1"])
+    def test_simple_tensor(self, p):
+        input = self.input_tensor()
+
+        actual = transforms.RandomHorizontalFlip(p=p)(input)
```

Comment:
Nit: although this is correct, I personally feel creating objects and directly calling them is somewhat hard to read. Maybe split this into two lines?

```suggestion
        transform = transforms.RandomHorizontalFlip(p=p)
        actual = transform(input)
```

No strong opinion though. Feel free to leave this open.